.vsts.pipelines/jobs/collect-artifacts.yml

```diff
@@ -0,0 +1,104 @@
+jobs:
+- job: Collect_Artifacts
+
+  dependsOn: ${{ parameters.dependsOn }}
+
+  displayName: Collect Artifacts
+
+  pool: ${{ parameters.pool }}
+
+  steps:
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download CentOS artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline
+        downloadPath: '$(Build.StagingDirectory)/centos'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Debian artifacts
+      inputs:
+        artifactName: Tarball debian9 Offline
+        downloadPath: '$(Build.StagingDirectory)/debian'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Fedora artifacts
+      inputs:
+        artifactName: Tarball fedora29 Offline
+        downloadPath: '$(Build.StagingDirectory)/fedora'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Ubuntu artifacts
+      inputs:
+        artifactName: Tarball ubuntu1604 Offline
+        downloadPath: '$(Build.StagingDirectory)/ubuntu'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download portable artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline Portable
+        downloadPath: '$(Build.StagingDirectory)/portable'
+
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        workingDirectory: '$(Build.StagingDirectory)'
+        script: |
+          version=`echo centos/*/Private.SourceBuilt.Artifacts*.tar.gz | sed 's/\.tar\.gz$//' | sed 's/^centos\/.*\/Private\.SourceBuilt\.Artifacts\.//'`
+          echo "Artifacts version: $version"
+          mkdir collected
+          cd collected
+          # do CentOS last to prefer those versions of any duplicates
+          for os in portable debian fedora ubuntu centos; do
+            # DownloadArtifact creates another directory - use '*' to not deal with this
+            tar xzf ../$os/*/Private.SourceBuilt.Artifacts*.tar.gz
+            rm -rf ../$os
+          done
+          echo $(Build.SourceVersion) > .version
+          echo $(Build.BuildNumber) > .buildnumber
+          echo $(Build.BuildId) > .buildid
+          tar czf ../Private.SourceBuilt.Artifacts.$version.tar.gz * .version .buildnumber .buildid
```

Comment:
This combines the nupkgs into the tarball, but doesn't combine the corresponding PackageVersions.props.  I'm not sure that it's an issue, though.

Reply:
I feel like we should do that but I don't think we currently have a use for it.  We also can work around it using something like
https://github.com/dotnet/source-build/blob/8d17a1046da907571c285ae144db4dc48ac9c9d2/dir.props#L211

Any opinion on whether we should fix this now or log an issue and do it if it ever comes up?

Reply:
I'm assuming that all of the packages will be the same name/version with the exception of the runtime packages that are rid specific.  I don't think these will use the version tag, so I think it will be OK to leave it.

Reply:
Sounds good for now, logged https://github.com/dotnet/source-build/issues/1423.